File: backend/app/models/category.py
from sqlalchemy import Column, DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import backref, relationship
from sqlalchemy.sql import func

from app.db.base import Base


class Category(Base):
    __tablename__ = "categories"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, index=True, nullable=False)
    description = Column(String, nullable=True)
    parent_id = Column(Integer, ForeignKey("categories.id"), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    # Self-referential relationship for hierarchical categories
    children = relationship(
        "Category",
        backref=backref("parent", remote_side=[id]),
        cascade="all, delete-orphan",
    )

    # Relationship with products
    products = relationship("Product", back_populates="category")

File: frontend/src/shared/api/types/Category.ts
export type Category = {
  name?: string | null;
  description?: string | null;
  parent_id?: number | null;
  /**
   * @type integer
   */
  id: number | null;
  /**
   * @type string, date-time
   */
  created_at?: string;
  updated_at?: string | null;
};
